iquip/apps/explorer.py

```diff
@@ -54,15 +56,36 @@ def loadFileTree(self):
         It assumes that all experiment files are in self.repositoryPath.
         """
         self.explorerFrame.fileTree.clear()
-        self._addFile(self.repositoryPath, self.explorerFrame.fileTree)
+        threading.Thread(
+            target=lambda: self._addFile(self.repositoryPath, self.explorerFrame.fileTree)
+        ).start()
+
+    @pyqtSlot(QTreeWidgetItem)
+    def lazyLoadFile(self, experimentFileItem: QTreeWidgetItem):
+        """Loads the experiment file in the directory.
+
+        This will be called when a directory item is expanded,
+        so it makes loading files lazy.
+
+        Args:
+            experimentFileItem: The expanded file item.
+        """
+        if experimentFileItem.childCount() != 1 or experimentFileItem.child(0).columnCount() != 0:
+            return
+        experimentFileItem.removeChild(experimentFileItem.child(0))
```

Comment:
Please add a comment to explain that there is an empty item for an unloaded directory.